milestones/services.py

```diff
@@ -0,0 +1,49 @@
+"""
+A wrapper class around all methods exposed in api.py
+"""
+
+import types
+
+
+class MilestonesService(object):
+    """
+    An xBlock service for xBlocks to talk to the Milestones subsystem.*
+
+    NOTE: This is a Singleton class. We should only have one instance of it!
+    """
+
+    _instance = None
+
+    REQUESTED_FUNCTIONS = [
+        'get_course_content_milestones',
+    ]
+
+    def __new__(cls, *args, **kwargs):
+        """
+        This is the class factory to make sure this is a Singleton
+        """
+        if not cls._instance:
+            cls._instance = super(MilestonesService, cls).__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self):
+        """
+        Class initializer, which just inspects the libraries and exposes the same functions
+        """
+        from util import milestones_helpers as milestones_api
+        self._bind_to_module_functions(milestones_api)
+
+    def _bind_to_module_functions(self, module):
```

Comment:
I wasn't sure which implementation was better from the two below (the commented one or the uncommented one). The uncommented one is more straightforward but it requires an additional function get_service(). Not sure if that matters.